backend/internal/service/email_service.go

```diff
@@ -1,62 +1,90 @@
 package service
 
 import (
+	"bytes"
 	"errors"
 	"fmt"
 	"github.com/stonith404/pocket-id/backend/internal/common"
+	"github.com/stonith404/pocket-id/backend/internal/utils/email"
+	htemplate "html/template"
+	"io/fs"
+	"mime/multipart"
+	"mime/quotedprintable"
 	"net/smtp"
-	"os"
-	"strings"
+	"net/textproto"
+	ttemplate "text/template"
 )
 
 type EmailService struct {
 	appConfigService *AppConfigService
+	htmlTemplates    map[string]*htemplate.Template
+	textTemplates    map[string]*ttemplate.Template
 }
 
-func NewEmailService(appConfigService *AppConfigService) *EmailService {
+func NewEmailService(appConfigService *AppConfigService, templateDir fs.FS) (*EmailService, error) {
```

Comment:
It would be definitely possible, but it makes the "service" with very simple functionality dependant on complex global variables. It would be much easier to test the templating alone without worrying about the whole global state. Also this allows the use of Golang embed, which could be useful in tests and if the app would be shipped as single binary.

I did a few bigger Golang projects and almost every time there was some shared global state, it led later to complications with testing isolated components. Personally, I think it is more readable, because I can see on the first glance, what the service is depending on.

Reply:
Makes completely sense. Thanks for the explanation.